examples/tv-app/android/java/ChannelManager.cpp

```diff
@@ -363,30 +371,273 @@ void ChannelManager::HandleGetProgramGuide(
+            if (jsubTitle != nullptr)
+            {
+                program->subtitle = MakeOptional(subTitle->charSpan());
+            }
+
+            jfieldID getDescriptionField = env->GetFieldID(programClass, "description", "Ljava/lang/String;");
+            jstring jdescription         = static_cast<jstring>(env->GetObjectField(programObject, getDescriptionField));
+            JniUtfString * description   = new JniUtfString(env, jdescription);
+            needToFreeStrings.push_back(description);
+            if (jdescription != nullptr)
+            {
+                program->description = Optional<CharSpan>(description->charSpan());
+            }
+
+            jfieldID startTimeField = env->GetFieldID(programClass, "startTime", "J");
+            jlong jstartTime        = env->GetLongField(programObject, startTimeField);
+            program->startTime      = static_cast<uint32_t>(jstartTime);
+
+            jfieldID endTimeField = env->GetFieldID(programClass, "endTime", "J");
+            jlong jendTime        = env->GetLongField(programObject, endTimeField);
+            program->endTime      = static_cast<uint32_t>(jendTime);
+
+            uint32_t recordFlag               = 0;
+            jfieldID recordFlagScheduledField = env->GetFieldID(programClass, "recordFlagScheduled", "Z");
+            jboolean jrecordFlagScheduled     = env->GetBooleanField(programObject, recordFlagScheduledField);
+            if (jrecordFlagScheduled)
+            {
+                recordFlag |= 1;
+            }
+            jfieldID recordFlagSeriesField = env->GetFieldID(programClass, "recordFlagSeries", "Z");
+            jboolean jrecordFlagSeries     = env->GetBooleanField(programObject, recordFlagSeriesField);
+            if (jrecordFlagSeries)
+            {
+                recordFlag |= 2;
+            }
+            jfieldID recordFlagRecordedField = env->GetFieldID(programClass, "recordFlagRecorded", "Z");
+            jboolean jrecordFlagRecorded     = env->GetBooleanField(programObject, recordFlagRecordedField);
+            if (jrecordFlagRecorded)
+            {
+                recordFlag |= 4;
+            }
+
+            program->recordingFlag = MakeOptional(static_cast<uint32_t>(recordFlag));
+
+            needToFreePrograms.push_back(program);
+            tempPrograms.push_back(*program);
+        }
+
+        response.programList = DataModel::List<const ProgramType>(tempPrograms.data(), tempPrograms.size());
+
+        err = helper.Success(response);
+
+        for (ProgramType * program : needToFreePrograms)
+        {
+            delete program;
+        }
+
+        for (JniUtfString * str : needToFreeStrings)
+        {
+            delete str;
+        }
+    }
+
+exit:
+    ChipLogProgress(Zcl, " -- HandleGetProgramGuide -- 55");
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "ChannelManager::HandleGetProgramGuide status error: %s", err.AsString());
```

Comment:
maybe it is better to use %" CHIP_ERROR_FORMAT "\n", err.Format()

Reply:
fixed - thanks!